packages/Php/tests/Rector/FuncCall/RegexDashEscapeRector/Fixture/fixture.php.inc

```diff
@@ -0,0 +1,33 @@
+<?php
+
+function regexDashEscapeRector()
+{
+    preg_match("#[\w()-]#", 'some text'); // ok
+    preg_match("#[-\w()]#", 'some text'); // ok
+    preg_match("#[\w-()]#", 'some text'); // NOPE!
+    preg_match("#[\w(-)]#", 'some text'); // ok
+
+    preg_match('#[\w()-]#', 'some text'); // ok
+    preg_match('#[-\w()]#', 'some text'); // ok
+    preg_match('#[\w-()]#', 'some text'); // NOPE!
+    preg_match('#[\w(-)]#', 'some text'); // ok
+}
+
+?>
+-----
+<?php
+
+function regexDashEscapeRector()
+{
+    preg_match("#[\w()-]#", 'some text'); // ok
+    preg_match("#[-\w()]#", 'some text'); // ok
+    preg_match("#[\w\-()]#", 'some text'); // NOPE!
+    preg_match("#[\w(-)]#", 'some text'); // ok
+
+    preg_match('#[\w()-]#', 'some text'); // ok
+    preg_match('#[-\w()]#', 'some text'); // ok
+    preg_match('#[\w\-()]#', 'some text'); // NOPE!
+    preg_match('#[\w(-)]#', 'some text'); // ok
+}
```

Comment:
@vrana Hi, we talked about regex on Zdroják.

Is this correct PHP 7.3 Preg regression replacement? Only 2 cases with `NOPE!` should be affected

It took me while you mean `-` as connecting 2 signs :)

---

This is the pattern I use:

```php
    /**
     * Matches:
     * [\w-\d]
     *
     * Skips:
     * [-
     * [\-
     * a-z
     * A-Z
     * 0-9
     * (-
     * -]
     * -)
     * @var string
     * @see https://regex101.com/r/6zvPjH/1/
     */
    private const PATTERN_DASH_NOT_AROUND_BRACKETS = '#(?<!\[|\(|A|a|0|\\\\)-(?!\]|\)|A|a|0)#';
```